test/BerfChatStorage.ts

```diff
@@ -175,12 +169,204 @@ describe("BerfChatStorage contract tests", async () => {
+
+
+        // Parse the string back to an object
+        const encryptedObject = EthCrypto.cipher.parse(encryptedString);
+
+        // Confirm that acountTres CANNOT decrypt message
+        await rejects(EthCrypto.decryptWithPrivateKey(accountTres.privateKey, encryptedObject));
+    });
+    */
+
+    it("tests sending a message via the contract", async () => {
+        // Declare and assign variable
+        // with secret string.
+        const secretMessage: string = "Satoshi is not Vitalik.";
+
+        const signature: string = EthCrypto.sign(
+            accountOnePrivKey,
+            EthCrypto.hash.keccak256(secretMessage)
+        );
+
+        
+        // Create object with message and signature
+        const payload: any = {
+            message: secretMessage,
+            signature
+        };
+        
+        // NOT SURE WHICH TYPE TO MAKE THIS
+        const encrypted: any = await EthCrypto.encryptWithPublicKey(
+            accountTwoPubKey,
+            JSON.stringify(payload)
+        );
+
+        // Convert object into smaller string-representation
+        // (This variable of type string)
+        const encryptedString: string = EthCrypto.cipher.stringify(encrypted);
+
+        // Confirm the sendMessage function emitted
+        // the MessageSent event
+        // Note: currently have to hardcode the Hardhat
+        // Network block number (parameter number 4)
+        await expect(berfChatStorage.sendMessage(accountTwo.address, encryptedString))
+        .to.emit(berfChatStorage, 'MessageSent')
+        .withArgs(
+            accountOne.address,
+            accountTwo.address,
+            (await berfChatStorage.hashAddresses(accountOne.address, accountTwo.address)),
+            encryptedString,
+            2 //Currently hard coding the HRE block number
+        );
+
+        // Parse the string back to an object
+        const encryptedObject: any = EthCrypto.cipher.parse(encryptedString);
+        
+        // Decrypt the received message with the
+        // private key of the recipient, accountTwo
+        const decrypted: string = await EthCrypto.decryptWithPrivateKey(
+            accountTwoPrivKey,
+            encryptedObject
+        );
+        
+        const decryptedPayload: any = JSON.parse(decrypted);
+        
+        // Check signature
+        const senderAddress: string = EthCrypto.recover(
+            decryptedPayload.signature,
+            EthCrypto.hash.keccak256(decryptedPayload.message)
+        );
+
+        // Confirm the address that sent it is accountOne
+        expect(senderAddress).to.equal(accountOne.address);
+
+        // Confirm the decrypted message is
+        // what was declared earlier
+        expect(decryptedPayload.message).to.equal(secretMessage);
+
+
+        // BELOW IS RESPONSE MESSAGE //
+
+
+        // Declare and assign variable
+        // with secret string.
+        const secretResponse: string = "Now I know Vitalik is not Satoshi.";
```

Comment:
best practice is to have a single unit tests covers a single use case

Reply:
I agree. I could split this one up in to two different unit tests, that cover the applications function from top to bottom.

Reply:
Quickly split up unit test into three different tests that check different functionality. Will clean up code further to apply DRY.